### PR TITLE
Init for backend switching action

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added `wait-until-build-succeeds` action to centralize waiting for build prow job.
- Added hello world for new backend switching job.

**Related issue(s)**

### CODE REVIEW

File: .github/workflows/e2e-k8s.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: e2e-on-k8s
```
Comment (marcobebway) [inline .github/workflows/e2e-k8s.yml:1]: why not just `e2e`?

Comment (mfaizanse) [reply .github/workflows/e2e-k8s.yml:1]: we have another workflow `e2e-without-lifecycle-manager`, which should be renamed to `e2e` IMO. Thats why kept the k8s to convey that this workflow uses real k8s on gardener.

File: .github/workflows/e2e-k8s.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+env:
+  KYMA_STABILITY: "unstable"
+  KYMA: "./hack/kyma"
+  MANAGER_IMAGE: europe-docker.pkg.dev/kyma-project/dev/eventing-manager:PR-${{ github.event.number }}
+
+on:
+  pull_request_target:
+    branches:
+      - main
+      - "release-*"
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply .github/workflows/e2e-k8s.yml:12]: I copied it from the other workflow. But I am sure atm, I will confirm this.

File: .github/workflows/e2e-k8s.yml (hunk 1 above)
Comment (marcobebway) [reply .github/workflows/e2e-k8s.yml:1]: How about we use the following prefixes when it make sense `e2e-k3e` `e2e-gardener`?

Comment (mfaizanse) [reply .github/workflows/e2e-k8s.yml:1]: I will do it in next PR.

**Pull request merged**